### PR TITLE
Draft Release 4: Basic Changes

### PR DESCRIPTION
**Pull request by KarstenSchnitter**
This is the first PR on the way to #164. It contains the following major changes:

- deletion of custom metrics feature
- removal of Jersey support and old sample app
- remove `javax.servlet` support in favour of Jakarta
- migrate to Java 17 (at Java 11 code level)
- upgrade to SLF4J 2.x and Logback/Log4j2 to latest versions
- upgrade JUnit and Mockito to v5
- replacement of Hamcrest by AssertJ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,11 +15,11 @@ git remote set-head origin -a
 
 ## Summary
 
-This is a collection of support libraries for Java applications (Java 8 and above) that serves three main purposes: 
+This is a collection of support libraries for Java applications (Java 11 and above) that serves three main purposes: 
 
 1. Provide means to emit *structured application log messages* 
 2. Instrument parts of your application stack to *collect request metrics* 
-3. Allow production of *custom metrics*.
+3. Allow auto-configuration of OpenTelemetry exporters.
```
Comment (nicklas-dohrn) [inline README.md:22]: really only auto configuration?

File: cf-java-logging-support-core/src/test/java/com/sap/hcp/cf/logging/common/EnvMap.java [hunk 2]
```diff
@@ -11,35 +11,8 @@ public class EnvMap {
...
-                                                  "}";
+    public static final String VCAP_APPLICATION =
+            "{" + "\"application_id\":\"" + VCAP_APP_ID + "\", " + "\"application_name\":\"" + VCAP_APP_NAME + "\", " + "\"application_uris\":[  " + "\"testapp.testdomain.com\" " + "], " + "\"application_version\":\"92720bd3-2735-489d-bf4b-e5bb05c752e7\", " + "\"cf_api\":\"https://api.cf.sap.hana.ondemand.com\", " + "\"host\":\"0.0.0.0\", " + "\"instance_id\":\"34b12c70-5443-426f-75c4-3260\", " + "\"instance_index\":\"" + VCAP_INSTANCE_IDX + "\", " + "\"limits\":{  " + "\"disk\":1024, " + "\"fds\":16384, " + "\"mem\":1024 " + "}, " + "\"name\":\"d048888-logging-sample-app\", " + "\"organization_id\":\"" + VCAP_ORGANIZATION_ID + "\", " + "\"organization_name\":\"" + VCAP_ORGANIZATION_NAME + "\", " + "\"port\":8080, " + "\"process_id\":\"a6026141-8d69-4e9d-b62d-8e1d982f7953\", " + "\"process_type\":\"web\", " + "\"space_id\":\"" + VCAP_SPACE_ID + "\", " + "\"space_name\":\"" + VCAP_SPACE_NAME + "\", " + "\"uris\":[  " + "\"testapp.testdomain.com\" " + "], " + "\"version\":\"92720bd3-2735-489d-bf4b-e5bb05c752e7\" " + "}";
```
Comment (nicklas-dohrn) [inline cf-java-logging-support-core/src/test/java/com/sap/hcp/cf/logging/common/EnvMap.java:16]: what is the benefit of "onelining" all this fields?

File: pom.xml [hunk 3]
```diff
@@ -24,76 +26,55 @@
...
-           <organization>SAP SE</organization>
-           <organizationUrl>https://github.com/SAP</organizationUrl>
-       </developer>
-       <developer>
-           <id>martein</id>
-           <name>Martin Nikolov</name>
-           <email>martin.nikolov@sap.com</email>
-           <organization>SAP SE</organization>
-           <organizationUrl>https://github.com/SAP</organizationUrl>
-       </developer>
```
Comment (nicklas-dohrn) [inline pom.xml:47]: some people are missing here, for example christian dinse.

**Review by nicklas-dohrn — APPROVED** [review]
looks good  as far as I can tell by looking briefly into all the changes.
Some minor remarks are added.

File: cf-java-logging-support-core/src/test/java/com/sap/hcp/cf/logging/common/EnvMap.java (hunk 2 above)
Comment (KarstenSchnitter) [reply cf-java-logging-support-core/src/test/java/com/sap/hcp/cf/logging/common/EnvMap.java:16]: No benefit. Automatic code-formatting.

File: README.md (hunk 1 above)
Comment (KarstenSchnitter) [reply README.md:22]: Almost. There is specific documentation on the component.

**Pull request merged**